File: java-micronaut-moshi/src/main/java/test_service/MoshiFactory.java
package test_service;

import com.squareup.moshi.Moshi;
import io.micronaut.context.annotation.*;
import jakarta.inject.Singleton;
import test_service.json.Json;

@Factory
public class MoshiFactory {
    @Singleton
    public Moshi getMoshi() {
        var moshiBuilder = new Moshi.Builder();
        Json.setupMoshiAdapters(moshiBuilder);
        return moshiBuilder.build();
    }
}

File: kotlin-moshi-micronaut/src/main/kotlin/test_service/MoshiFactory.kt
package test_service

import com.squareup.moshi.Moshi
import io.micronaut.context.annotation.Factory
import jakarta.inject.Singleton
import test_service.json.setupMoshiAdapters

@Factory
class MoshiFactory {
    @Singleton
    fun getMoshi(): Moshi {
        val moshiBuilder = Moshi.Builder()
        setupMoshiAdapters(moshiBuilder)
        return moshiBuilder.build()
    }
}